apps/omg_api/test/support/state/prop_test/everyone_exit.ex

```diff
@@ -0,0 +1,51 @@
+# Copyright 2018 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.API.State.PropTest.EveryoneExit do
+  @moduledoc """
+  Generates function needed to get everyone out of their utxo
+  """
+  use PropCheck
+  alias OMG.API.PropTest.Helper
+  alias OMG.API.Utxo
+  require Utxo
+  def impl(exiting_utxos), do: OMG.API.State.PropTest.StateCoreGS.exit_utxos(exiting_utxos)
+
+  def args(%{model: %{history: history}}),
+    do: [
+      Helper.spendable(history)
+      |> Enum.map(fn {{blknum, txindex, oindex}, %{owner: owner}} ->
+        %{utxo_pos: Utxo.Position.encode(Utxo.position(blknum, txindex, oindex)), owner: Helper.get_addr(owner)}
+      end)
+    ]
+
+  def post(_, _, {:ok, _}), do: true
+
+  def next(%{model: %{history: history, balance: balance} = model} = state, [exits], _) do
+    delete_utxo =
+      exits
+      |> Enum.map(fn %{utxo_pos: position} ->
+        {:utxo_position, blknum, txindex, oindex} = Utxo.Position.decode(position)
```

Comment:
oh didn't notice before. Wouldn't you like to store the `Utxo.Position` struct, instead of the tuple?

You can pattern match `Utxo.position(blknum, txindex, oindex) = utxo_position` using the `position` macro.

Places like https://github.com/omisego/elixir-omg/pull/160/files#diff-3e9cfd586ad9c4b216dcaff630158afcR28 would be a lot simpler. https://github.com/omisego/elixir-omg/pull/160/files#diff-3e9cfd586ad9c4b216dcaff630158afcR38 too. Just use the struct everywhere